traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/router/TrafficRouter.java

```diff
@@ -893,12 +978,19 @@ public Cache consistentHashForGeolocation(final String ip, final String delivery
 			return null;
 		}
 
-		final String pathToHash = buildPatternBasedHashString(deliveryService, requestPath);
+		final String pathToHash = buildPatternBasedHashString(deliveryService, request);
 		return consistentHasher.selectHashable(caches, deliveryService.getDispersion(), pathToHash);
 	}
 
 	public String buildPatternBasedHashStringDeliveryService(final String deliveryServiceId, final String requestPath) {
-		return buildPatternBasedHashString(cacheRegister.getDeliveryService(deliveryServiceId), requestPath);
+		final HTTPRequest r = new HTTPRequest();
+		r.setPath(requestPath);
+		r.setQueryString("");
+		return buildPatternBasedHashString(cacheRegister.getDeliveryService(deliveryServiceId), r);
+	}
+
+	public String buildPatternBasedHashStringDeliveryService(final String deliveryServiceId, final HTTPRequest request) {
```

Comment:
My IDE is flagging this method as never used. Can it be deleted?

Reply:
same as above